kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -290,101 +277,103 @@ object Result:
           * @return
           *   true if the Result is a Success, false otherwise
           */
-        def isSuccess: Boolean =
-            self match
-                case _: Error[?] => false
-                case _           => true
+        def isSuccess: Boolean = !isError
```

Comment:
More consistent API pattern with operations for success, generic error, and specific failure/panic